src/process_watch.js

```diff
@@ -0,0 +1,369 @@
+      return ((currentProcess.pid == ele.pid) && (currentProcess
+        .cpu <=
+        cpuRemoveLevel));
+    };
+    return processes.some(findCPU);
+  });
+  _.remove(processWarnList, function (ele) {
+    var findCPU = function (currentProcess) {
+      //matches element && cpu is below removeLevel
+      return ((currentProcess.pid == ele.pid) && (currentProcess
+        .cpu <
+        cpuRemoveLevel));
+    };
+    return processes.some(findCPU);
+  });
+
+}
+/**
+ * toMinutes
+ * @param {number} miliseconds
+ * the difference between two timestamps 
+ * int to be converted to a human readable number
+ */
+function toMinutes(millis) {
+  var minutes = Math.floor(millis / 60000);
+  var seconds = ((millis % 60000) / 1000).toFixed(0);
+  return minutes + "." + (seconds < 10 ? '0' : '') + seconds;
+}
+/**
+ * formatBytes
+ * @param {number, null} bytes, (option to be more precise)
+ * int to be converted to a bytes string. 
+ */
+function formatBytes(bytes, decimals) {
+  if (bytes == 0) return '0 Bytes';
+  var k = 1024,
+    dm = decimals <= 0 ? 0 : decimals || 2,
+    sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'],
+    i = Math.floor(Math.log(bytes) / Math.log(k));
+  return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+}
+/**
+ * recordWarnProcess
+ * @param {obj} processObject
+ * the report object sent by checkProcess.
+ * @param time currentTime
+ * the current time, sent by checkProcess.
+ */
+function recordWarnProcess(processObject, currentTime) {
+  //currentTime = new Date().getTime();
+  var pidStored = function (stored) {
+    if (stored == undefined) {
+      return false;
+    }
+    return processObject.pid == stored.pid;
+  };
+  let found = 0;
+  pidusage(processObject.pid, function (err, stats) {
+    // all processes always get updated with this information
+    processObject.elapsed = toMinutes(stats.elapsed);
+    processObject.memory = formatBytes(stats.memory);
+
+  })
+  if (processWarnList.some(pidStored)) {
+    processObject.lastDetectTime = currentTime;
+    //if exists in storage
+    // update; cpu, memory, lastDetectTime, 
+
+    //element.lastDetectTime <= (currentTime - reportInterval)
+    index = _.findIndex(processWarnList, function (element) {
+      return element.pid == processObject.pid
+    })
+    /* 
+     * found time, Show how long a ps has been on list
+     * age should be difference between found and current time 
+     */
+    processObject.firstDetectTime = processWarnList[index].firstDetectTime;
+    // reportTime, Show how long since last report
+    processObject.reportTime = processWarnList[index].reportTime;
+    // If a new process, that is repeat offender, allow reporting it
+    // Also a little over 15 min old 
+    processObject.report = ((processObject.reportTime == undefined) && (
+      processObject.firstDetectTime <= (currentTime - 15 * 60 * 1000)));
+    //overwrite previous
+    processWarnList.splice(index, 1, processObject);
```

Comment:
so i'm curious, why we are creating a new object, then overwriting it in the processWarnList, instead of just updating the existing entry? You've already got the index, why not just update the entry with the updated values?

Reply:
Originally it had something to do with how **finding** that existing entry took too long, so creating a new object to insert worked better.
Eventually changed the action order, but didn't take that out